datarobot_provider/operators/ai_catalog.py

```diff
@@ -59,3 +59,61 @@ def execute(self, context: Dict[str, Any]) -> str:
         ai_catalog_dataset = dr.Dataset.create_from_file(context["params"]["dataset_file_path"])
         self.log.info(f"Dataset created: dataset_id={ai_catalog_dataset.id}")
         return ai_catalog_dataset.id
+
+
+class UpdateDatasetFromFileOperator(BaseOperator):
+    """
+    Operator that creates a new Dataset version from a file.
+    Returns when the new dataset version has been successfully uploaded.
+
+    :param dataset_id: DataRobot AI Catalog dataset ID
+    :type dataset_id: str, optional
+    :param datarobot_conn_id: Connection ID, defaults to `datarobot_default`
+    :type datarobot_conn_id: str, optional
+    :return: DataRobot AI Catalog dataset version ID
+    :rtype: str
+    """
+
+    # Specify the arguments that are allowed to parse with jinja templating
+    template_fields: Iterable[str] = ["dataset_id"]
+    template_fields_renderers: Dict[str, str] = {}
+    template_ext: Iterable[str] = ()
+    ui_color = '#f4a460'
+
+    def __init__(
+        self,
+        *,
+        dataset_id: str = None,
+        datarobot_conn_id: str = "datarobot_default",
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.dataset_id = dataset_id
+        self.datarobot_conn_id = datarobot_conn_id
+        if kwargs.get('xcom_push') is not None:
+            raise AirflowException(
+                "'xcom_push' was deprecated, use 'BaseOperator.do_xcom_push' instead"
+            )
+
+    def execute(self, context: Dict[str, Any]) -> str:
+        # Initialize DataRobot client
+        DataRobotHook(datarobot_conn_id=self.datarobot_conn_id).run()
+
+        # If dataset_id not provided in constructor, then using training_dataset_id from json config
+        dataset_id = (
+            self.dataset_id
+            if self.dataset_id is not None
+            else context['params']['training_dataset_id']
+        )
```

Comment:
What is the usecase for these different code paths, e.g. setting the ID from the constructor vs inferring it from the context -- because the file_path is only set from the context so why only the dataset_id can be set in two different ways?